src/index.ts

```diff
@@ -142,6 +163,24 @@ function getMappedPath(source: string, file: string) {
   return paths[0]
 }
 
+/**
+ * Like `createMatchPath` from `tsconfig-paths` package, but considers
+ * that the module id could have a .js or .jsx extension.
+ */
+const createExtendedMatchPath: typeof createMatchPath = (
+  absoluteBaseUrl,
+  paths,
+  ...rest
+) => {
+  const matchPath = createMatchPath(absoluteBaseUrl, paths, ...rest)
+
+  return (id, ...otherArgs) => {
+    const match = matchPath(id, ...otherArgs)
+    if (match != null) return match
+    return matchPath(removeJsExtension(id), ...otherArgs)
```

Comment:
Check `idWithoutJsExt !== id` like above.

Reply:
fixed in https://github.com/alexgorbatchev/eslint-import-resolver-typescript/pull/56/commits/54eb91b4faff1bbf310c390fd554e3990844d4a6